cinema/views.py

```diff
@@ -24,21 +27,53 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed for this resource")
+        return [permission() for permission in permission_classes]
 
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed for this resource")
```

Comment:
Raising a `NotFound` exception for unsupported actions is misleading. Consider using `MethodNotAllowed` to indicate that the HTTP method is not allowed for the requested resource.